### PR TITLE
LPS76201: Use Gulp for Metal and Module Config Invocation

### PR DESCRIPTION
**Pull request by christopherbryanboyd**
This is to fix IDE-3359: Maven-Soy Infinite Build Loop in Eclipse.

This uses the **frontend-maven-plugin** to invoke **gulp**, which then invokes the **metal-cli** and **liferay-module-config-generator** node modules, and includes its own _M2E_ integration which is only triggered when resources in the relevant directory (_src/main/resources/META-INF/resources/_) are touched.

### CODE REVIEW

**Comment by christopherbryanboyd** [comment]
ci:test

File: modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
 				<include>bnd.bnd</include>
 				<include>build.gradle</include>
 				<include>package.json</include>
+				<include>gulpfile.js</include>
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/META-INF/maven/archetype-metadata.xml:24]: Sort the include

File: modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
 	},
 	"devDependencies": {
 		"liferay-module-config-generator": "^1.2.1",
-		"metal-cli": "^4.0.1"
+		"metal-cli": "^4.0.1",
+		"gulp": "latest"
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/package.json:9]: Lets lock down a gulp to specific version.

Comment (gamerson) [reply modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/package.json:9]: Look at what the other node themes use for their gulp version.

File: modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+var gulp = require('gulp');
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/gulpfile.js:1]: Source formatter?  Can you run it on this file?

File: modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml [hunk 4]
```diff
@@ -66,74 +66,24 @@
...
-				<executions>
-					<execution>
+                    <execution>
 						<id>transpile-js</id>
-						<phase>process-classes</phase>
 						<goals>
-							<goal>exec</goal>
+							<goal>gulp</goal>
 						</goals>
+						<phase>process-classes</phase>
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml:74]: Lets put the phase above goals even though it isn't alphabetic.

File: modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml [hunk 5]
```diff
@@ -217,41 +167,5 @@
...
-							</pluginExecutions>
-						</lifecycleMappingMetadata>
-					</configuration>
-				</plugin>
-			</plugins>
-		</pluginManagement>
 	</build>
-</project>
\ No newline at end of file
+</project>
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml:171]: remove extra newline

File: modules/sdk/project-templates/project-templates/build.gradle [hunk 6]
```diff
@@ -176,4 +176,4 @@ writeProjectTemplateVersions {
 			properties.store it, null
 		}
 	}
-}
\ No newline at end of file
+}
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates/build.gradle:179]: remove extra newline

**Comment by gamerson** [comment]
Also can you add some new unit test that asserts this new gulpfile exists.

**Pull request closed without merging**